### ISSUE TITLE
Make it easier to report errors from `preProcess`

### ISSUE THREAD

**Issue by jyasskin** [w3c/respec#4560]
**Is your feature request related to a problem? Please describe.**

In https://github.com/w3ctag/privacy-principles/pull/355, I added `data-algorithm` attributes to each [practice](https://respec.org/docs/#practice) to support some future UI, and I wrote a `preProcess` function to check that we're using the attribute consistently. If we're not using it right, I want to report an error, but the only way I saw to report that error was to throw an exception. This exception shows up in the UI and spec-prod GH Action as just ["Function checkAudiences threw an error during preProcess."](https://github.com/w3ctag/privacy-principles/actions/runs/6485892952/job/17612929653), which isn't very helpful.

**Describe the solution you'd like**

I should be able to call some function to add to Respec's native errors list.

**Comment by marcoscaceres** [comment]
Yeah, we could expose src/core/utils.js 's `showError()` and `showWarning()` on `document.respec`. 


That could be nice.

**Comment by marcoscaceres** [comment]
@jyasskin, or anyone willing to take this on... it should be really easy:

In src/core/respec-global.js:

1. import showError() and showWarning() from core/utils
1.  add a showError() and showWarning() to class ReSpec, recycling the showError, and showWarning function signatures.
3. add a couple of tests to where respec-global is tested... there are tests already for showError and showWarning, which can be copy/pasted.

**Comment by marcoscaceres** [comment]
Sure, that seems good too.
